### PR TITLE
Tup#242

### PR DESCRIPTION
**Pull request by craxrev**
Introduces the new payment feature

### CODE REVIEW

File: controllers/OverviewController.php [hunk 1]
```diff
@@ -45,4 +48,35 @@ public function actionShareholderList()
     {
         return $this->render('shareholder-list', ['asset' => AssetHelper::getSpaceAsset($this->contentContainer)]);
     }
+
+    public function actionPurchaseCoin()
+    {
+        $amountModel = new DynamicModel(['amount']);
```
Comment (gdalyy) [inline controllers/OverviewController.php:54]: Why not creating a dedicated form ? maybe this DynamicModel will containg some other fields in upcoming updates ?

File: controllers/OverviewController.php [hunk 2]
```diff
@@ -45,4 +48,35 @@ public function actionShareholderList()
...
+        $amountModel->addRule(['amount'], 'integer', ['min' => 0]);
+        $amountModel->addRule(['amount'], 'required');
+
+        if ($amountModel->load(Yii::$app->request->post()) && $amountModel->validate()) {
+            $bridge = Yii::$app->params['coinPurchase']['bridge'];
+            $defaultAccount = Account::findOne([
+                'user_id' => $this->contentContainer->id,
+                'account_type' => Account::TYPE_DEFAULT
+            ]);
+            $data = [
```
Comment (gdalyy) [inline controllers/OverviewController.php:64]: new line here

File: controllers/OverviewController.php [hunk 3]
```diff
@@ -45,4 +48,35 @@ public function actionShareholderList()
...
+                'fullName' => $this->contentContainer->displayName,
+                'coin' => 'coinsence',
+                'amount' => intval($amountModel->amount),
+                'pAddress' => $defaultAccount->ethereum_address,
+                'rediectUrl' => Url::toRoute(['/xcoin/overview', 'contentContainer' => $this->contentContainer], true) . '?res=success',
+            ];
+            $jsonData = json_encode($data);
```
Comment (gdalyy) [inline controllers/OverviewController.php:71]: new line here too

**Review by gdalyy — CHANGES_REQUESTED** [review]

File: controllers/OverviewController.php (hunk 1 above)
Comment (craxrev) [reply controllers/OverviewController.php:54]: Was thinking not worth doing a full Form for just one field, but I'll make one if more fields should come

**Review by gdalyy — APPROVED** [review]

**Pull request merged**